### ISSUE TITLE
Always open this site in a specific container causes double tab opened

### ISSUE THREAD

**Issue by PhilipDukhov** [chronakis/firefox-sticky-window-containers#15]
**Describe the bug**
For some sites, like GitHub, I don't want to use different accounts in different containers. I'm trying to solve this by using **Always open this site** option of **Multi-Account Containers**, but this makes the bookmark opened twice.

**To Reproduce**
1. Mark site as `Always open this site`
2. Open it from a bookmark
3. For the first time, it asks whether I wanna to always open this site in this container. Select **Remember** checkmark and agree.
4. Since then always when I open a bookmark, two tabs appears.

**Expected behavior**
I expect one tab to be opened in remembered container no matter which container is the first tab of this window is. Or open it in the window with this container tabs and switch to it.

**Screenshots**

![res](https://user-images.githubusercontent.com/6103621/141599854-a1e61226-8407-4e25-a2f2-533009a0fefb.gif)

**Desktop (please complete the following information):**
 - OS: MacOS
 - Browser Firefox 
 - Version 94.0.1

**Comment by chronakis** [comment]
Hi @PhilipDukhov,

Thank you for reporting this. It took me a while to see the report.

I managed to replicate with a few extra conditions:
* The first tab on the window has to be of the same container as the one the bookmark is set to open
* The bookmark has to open with middle click/command click/open in in tab
For me, every other sitatuation worked correctly but I think it is a time related issue so experience may vary. It happens ~7/10 times.

I will try to figure out what happens and fix it.

Thanks

**Comment by eppfel** [comment]
Yeah, this is making the extension unusable for me. I have quite strict rules for my containers, e.g. Reddit always opening in a personal container or my bank in the banking container, but for those two tabs are opening.

**Comment by eppfel** [comment]
Investigated a bit in the debugger, but it seems to happen during some async operation. I guess that the firefox container extension itself creates a new tab in the required context, which then gets added to the processing. 

Before https://github.com/chronakis/firefox-sticky-window-containers/blob/5c2abde7f865648e71d0b3c69ae5c350875107be/swc-background.js#L80 there is one tab but once `.then(` is called there are multiple.

**Comment by the21st** [comment]
This also happens to me. Would be great to get this fixed <3

**Comment by eppfel** [comment]
I switched to https://github.com/menhera-org/TabArray, which has much more features that I don't all use, but it keeps tab groups perfectly together.

**Comment by zlee-personal** [comment]
This can be fixed by checking if there is already a container assignment from the multi account containers add-on via browser.runtime.sendMessage(). See https://github.com/tiansh/always-in-container/tree/master. I can submit PR later or if @chronakis  wants to.

```
const isMACAssigned = async function (url) {
    if (!macAddonEnabled) return false;

    try {
      const assignment = await browser.runtime.sendMessage(MAC_ADDON_ID, {
        method: 'getAssignment',
        url,
      });
      return Boolean(assignment);
    } catch (e) {
      return false;
    }
  };
```

**Comment by chronakis** [comment]
Thank you @zlee-personal , I will free some time to look at it over the weekend.
